ts/src/bydfi.ts

```diff
@@ -0,0 +1,2954 @@
+        const marketId = this.safeString (trade, 'symbol');
+        market = this.safeMarket (marketId, market);
+        const timestamp = this.safeInteger (trade, 'time');
+        let fee = undefined;
+        const rawType = this.safeString (trade, 'type');
+        const feeCost = this.safeString (trade, 'fee');
+        if (feeCost !== undefined) {
+            fee = {
+                'cost': feeCost,
+                'currency': undefined,
+            };
+        }
+        const orderId = this.safeString (trade, 'orderId');
+        let side: Str = undefined; // fetchMyTrades always returns side BUY
+        if (orderId === undefined) {
+            // from fetchTrades
+            side = this.safeStringLower (trade, 'side');
+        }
+        return this.safeTrade ({
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': market['symbol'],
+            'id': this.safeString (trade, 'id'),
+            'order': orderId,
+            'type': this.parseTradeType (rawType),
+            'side': side,
+            'takerOrMaker': undefined,
+            'price': this.safeString2 (trade, 'price', 'dealPrice'),
+            'amount': this.safeString2 (trade, 'quantity', 'dealVolume'),
+            'cost': undefined,
+            'fee': fee,
+        }, market);
+    }
+
+    parseTradeType (type: Str): Str {
+        const types = {
+            '1': 'limit',
+            '2': 'market',
+            '3': 'liquidation',
+        };
+        return this.safeString (types, type, type);
+    }
+
+    /**
+     * @method
+     * @name bydfi#fetchOHLCV
+     * @description fetches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+     * @see https://developers.bydfi.com/en/swap/market#candlestick-data
+     * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+     * @param {string} timeframe the length of time each candle represents
+     * @param {int} [since] timestamp in ms of the earliest candle to fetch
+     * @param {int} [limit] the maximum amount of candles to fetch (max 500)
+     * @param {object} [params] extra parameters specific to the bitteam api endpoint
+     * @param {int} [params.until] timestamp in ms of the latest candle to fetch
+     * @returns {int[][]} A list of candles ordered as timestamp, open, high, low, close, volume
+     */
+    async fetchOHLCV (symbol: string, timeframe = '1m', since: Int = undefined, limit: Int = undefined, params = {}): Promise<OHLCV[]> {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const maxLimit = 500; // docs says max 1500, but in practice only 500 works
+        const interval = this.safeString (this.timeframes, timeframe, timeframe);
+        const request = {
+            'symbol': market['id'],
+            'interval': interval,
+        };
+        let paginate = false;
```

Comment:
I would move the pagination logic right below loadMarkets() to make it consistent.